src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -981,11 +981,24 @@ void LedgerConsensusImp::accept (std::shared_ptr<SHAMap> set)
         auto buildLCL = std::make_shared<Ledger>(
             *mPreviousLedger,
             app_.timeKeeper().closeTime());
-        auto constexpr v2_ledger_seq_switch = 40'000'000;
-        if (buildLCL->info().seq > v2_ledger_seq_switch &&
-           !buildLCL->stateMap().is_v2())
+        auto const v2_enabled = buildLCL->rules().enabled(featureSHAMapV2,
+                                                       app_.config().features);
+        auto v2_transition = false;
+        if (v2_enabled && !buildLCL->stateMap().is_v2())
         {
-            buildLCL->make_v2();
+            if (mHaveCorrectLCL)
+            {
+                int txCount = 0;
+                for (auto& txn : *set)
+                {
+                    if (++txCount > 10)
+                    {
+                        buildLCL->make_v2();
+                        v2_transition = true;
+                        break;
+                    }
```

Comment:
This code should be removed and changed back to the regular code. It was only used to make it easier to reproduce a bug by making the cutover ledger have lots of transactions to stress the cutover code.